scapy/layers/inet.py

```diff
@@ -853,13 +853,21 @@ class IPerror(IP):
     def answers(self, other):
         if not isinstance(other, IP):
             return 0
-        if not (((conf.checkIPsrc == 0) or (self.dst == other.dst)) and
-                (self.src == other.src) and
-                (((conf.checkIPID == 0) or
-                  (self.id == other.id) or
-                  (conf.checkIPID == 1 and self.id == socket.htons(other.id)))) and  # noqa: E501
-                (self.proto == other.proto)):
+
+        # Check if IP addresses match
+        test_IPsrc = conf.checkIPsrc == 0 or self.src == other.src
```

Comment:
I'd rather have the boolean values tested as boolean:
```suggestion
        test_IPsrc = not conf.checkIPsrc or self.src == other.src
```